### PR TITLE
Functionality to enable/disable trace logging

### PR DESCRIPTION
**Pull request by StanleyGoldman**
- Added a setting to enable/disable trace logging
- Added functionality to control `Verbose` output in `Serilog`
- Configured `Serilog` file output to allow writing from multiple processes
- Configured the log format to show the process id

Depends on
- [x] #1429 
- [x] #1428

### CODE REVIEW

File: src/GitHub.Exports/Settings/generated/IPackageSettings.cs [hunk 1]
```diff
@@ -31,6 +31,7 @@ public interface IPackageSettings : INotifyPropertyChanged
         void Save();
         bool CollectMetrics { get; set; }
         bool EditorComments { get; set; }
+        bool EnableTraceLogging { get; set; }
```
Comment (grokys) [inline src/GitHub.Exports/Settings/generated/IPackageSettings.cs]: As the header suggests, this is a generated file so you should be editing `src/Common/settings.json` to modify this.

I don't really like this, but that's how it is for the moment :(

File: src/GitHub.VisualStudio/Services/GitHubServiceProvider.cs [hunk 2]
```diff
@@ -97,16 +97,23 @@ public GitHubServiceProvider(IServiceProviderPackage asyncServiceProvider, IServ
...
-            instance = sp.GetService(serviceType);
-            if (instance != null)
-                return instance;
+            try
+            {
+                instance = sp.GetService(serviceType);
+                if (instance != null)
+                    return instance;
+            }
+            catch (Exception ex)
```
Comment (grokys) [inline src/GitHub.VisualStudio/Services/GitHubServiceProvider.cs]: This seems unrelated to trace logging. No objection to including it here, but could you explain how it got here? :D

File: src/GitHub.VisualStudio/Settings/generated/PackageSettingsGen.cs [hunk 3]
```diff
@@ -43,6 +43,13 @@ public bool CollectMetrics
             set { collectMetrics  = value; this.RaisePropertyChange(); }
         }
 
+        bool enableTraceLogging;
```
Comment (grokys) [inline src/GitHub.VisualStudio/Settings/generated/PackageSettingsGen.cs]: Again, this is a generated file so shouldn't be editing it directly.

File: src/GitHub.VisualStudio/Settings/PackageSettings.cs [hunk 4]
```diff
@@ -10,13 +11,61 @@
 namespace GitHub.VisualStudio.Settings
 {
     [Export(typeof(IPackageSettings))]
-    [PartCreationPolicy(CreationPolicy.Shared)]
+    [PartCreationPolicy(CreationPolicy.NonShared)]
+    public class PackgeSettingsDispatcher : IPackageSettings
```
Comment (grokys) [inline src/GitHub.VisualStudio/Settings/PackageSettings.cs]: I think we can use the technique from #1518 for this too.

File: src/GitHub.VisualStudio/UI/Settings/OptionsControl.xaml [hunk 5]
```diff
@@ -81,6 +81,22 @@
       </TextBlock>
       </DockPanel>
     </GroupBox>
+    <GroupBox
+            Margin="8,0,8,0"
+            VerticalAlignment="Top"
+            BorderBrush="LightGray"
```
Comment (grokys) [inline src/GitHub.VisualStudio/UI/Settings/OptionsControl.xaml]: Not sure why this `<WrapPanel>` is necessary seeing as there is only one child (a `WrapPanel` with one child won't do anything as far as I understand) but I see the other sections are the same so /shrug.

File: src/GitHub.VisualStudio/GitHubPackage.cs [hunk 6]
```diff
@@ -39,6 +41,19 @@ public class GitHubPackage : AsyncPackage
         {
             LogVersionInformation();
             await base.InitializeAsync(cancellationToken, progress);
+            
+            var packageSettings = await GetServiceAsync(typeof(IPackageSettings)) as IPackageSettings;
```
Comment (grokys) [inline src/GitHub.VisualStudio/GitHubPackage.cs]: Could we move this into an `InitializeLogging` method or something?

File: src/GitHub.Exports/Settings/generated/IPackageSettings.cs (hunk 1 above)
Comment (StanleyGoldman) [reply src/GitHub.Exports/Settings/generated/IPackageSettings.cs]: I can't believe i missed that..

File: src/GitHub.VisualStudio/Services/GitHubServiceProvider.cs (hunk 2 above)
Comment (StanleyGoldman) [reply src/GitHub.VisualStudio/Services/GitHubServiceProvider.cs]: I'm unable to reproduce the error @shana and I saw that caused us to add this catch.

File: src/GitHub.VisualStudio/Settings/generated/PackageSettingsGen.cs (hunk 3 above)
Comment (StanleyGoldman) [reply src/GitHub.VisualStudio/Settings/generated/PackageSettingsGen.cs]: Cool

File: src/GitHub.VisualStudio/Settings/PackageSettings.cs (hunk 4 above)
Comment (StanleyGoldman) [reply src/GitHub.VisualStudio/Settings/PackageSettings.cs]: 👍

File: src/GitHub.VisualStudio/UI/Settings/OptionsControl.xaml (hunk 5 above)
Comment (StanleyGoldman) [reply src/GitHub.VisualStudio/UI/Settings/OptionsControl.xaml]: It does not seem to have an effect on this element.

File: src/GitHub.VisualStudio/GitHubPackage.cs (hunk 6 above)
Comment (StanleyGoldman) [reply src/GitHub.VisualStudio/GitHubPackage.cs]: 👍

File: src/GitHub.VisualStudio/Services/GitHubServiceProvider.cs (hunk 2 above)
Comment (StanleyGoldman) [reply src/GitHub.VisualStudio/Services/GitHubServiceProvider.cs]: Should I remove the try/catch?

**Comment by StanleyGoldman** [comment]
@grokys I got to most of the fixes you mentioned.

File: src/GitHub.VisualStudio/Services/GitHubServiceProvider.cs (hunk 2 above)
Comment (grokys) [reply src/GitHub.VisualStudio/Services/GitHubServiceProvider.cs]: No, leave it in, it makes sense to log exceptions here. I just wondered what caused you to add it.
